CRM/Core/DomainTokens.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+use Civi\Token\AbstractTokenSubscriber;
+use Civi\Token\TokenRow;
+
+/**
+ * Class CRM_Case_Tokens
+ *
+ * Generate "case.*" tokens.
+ */
+class CRM_Core_DomainTokens extends AbstractTokenSubscriber {
+  /**
+   * @var string
+   *   Token prefix
+   */
+  public $entity = 'domain';
+
+  /**
+   * @var array
+   *   List of tokens provided by this class
+   *   Array(string $fieldName => string $label).
+   */
+  public $tokenNames;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct() {
+    parent::__construct($this->entity, $this->getDomainTokens());
+  }
+
+  public function getDomainTokens(): array {
+    return [
+      'name' => ts('Domain name'),
+      'address' => ts('Domain (organization) address'),
+      'phone' => ts('Domain (organization) phone'),
+      'email' => ts('Domain (organization) email'),
+      'id' => ts('Domain ID'),
+      'description' => ts('Domain Description'),
+    ];
+  }
+
+  /**
+   * @inheritDoc
+   * @throws \CRM_Core_Exception
+   */
+  public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL): void {
+    $row->format('text/html')->tokens($entity, $field, self::getDomainTokenValues()[$field]);
+    $row->format('text/plain')->tokens($entity, $field, self::getDomainTokenValues(NULL, FALSE)[$field]);
+  }
+
+  /**
+   * Get the tokens available for the domain.
+   *
+   * This function will be made protected soon...
+   *
+   * @param int|null $domainID
+   * @param bool $html
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   * @internal
+   *
+   */
+  public static function getDomainTokenValues(?int $domainID = NULL, bool $html = TRUE): array {
```

Comment:
this function is moved but otherwise unchanged by this PR - it was recently extracted from a bigger function